src/hotspot/share/code/debugInfo.cpp

```diff
@@ -191,6 +216,104 @@ void ObjectValue::print_fields_on(outputStream* st) const {
 #endif
 }
 
+
+// ObjectMergeValue
+
+// Returns the ObjectValue that should be used for the local that this
+// ObjectMergeValue represents. ObjectMergeValue represents allocation
+// merges in C2. This method will select which path the allocation merge
+// took during execution of the Trap that triggered the rematerialization
+// of the object.
+ObjectValue* ObjectMergeValue::select(frame* fr, RegisterMap* reg_map) {
+  assert(fr != nullptr && reg_map != nullptr, "sanity");
+
+  // If we call select again on the same merge we should return the same result
+  if (_selected != nullptr) {
+    return _selected;
```

Comment:
I'm not sure I understand how it is intended to work. The code below initializes `_selected`, but returns `nullptr` when `selector >= 0`. Subsequent calls will return non-null value.

Reply:
This can be improved. I'll fix it.